### PR TITLE
ci: use k8s pods

### PR DESCRIPTION
**Pull request by v1v**
PoC regarding k8s pods

### Issues

- https://github.com/elastic/apm-pipeline-library/issues/1552
- NODE_LABELS

### Blockers

- make: not found in the existing docker image, so we might need to build our own docker image with the toolchain

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -547,3 +551,13 @@ pipeline {
     }
   }
 }
+
+
+def runInPod(version, Closure body) {
+  container(version.replaceAll('\\.', '-')) {
```
Comment (kuisathaverat) [inline Jenkinsfile:557]: not needed you can set the default container to the Go container

File: Jenkinsfile [hunk 2]
```diff
@@ -107,10 +109,10 @@ pipeline {
       }
       steps {
         withGithubNotify(context: 'Intake') {
-          deleteDir()
-          unstash 'source'
-          dir("${BASE_DIR}"){
-            withGoEnv(){
+          runInPod(env.GO_VERSION) {
```
Comment (kuisathaverat) [inline Jenkinsfile:112]: you can interpolate variables in the pod definition, so you can launch small pods

**Comment by simitt** [comment]
@v1v is this something that you are going to be picking up again or should we close this for now?

**Comment by v1v** [comment]
> @v1v is this something that you are going to be picking up again or should we close this for now?

@simitt , we are working on this but in a best effort, let me close this as I cannot say when I'll pick this up yet

**Pull request closed without merging**